sw/device/lib/dif/dif_pinmux.c

```diff
@@ -282,9 +282,9 @@ dif_result_t dif_pinmux_pad_write_attrs(const dif_pinmux_t *pinmux,
 
   uint32_t read_value = mmio_region_read32(pinmux->base_addr, reg_offset);
   *attrs_out = dif_pinmux_reg_to_pad_attr(read_value);
-  if (reg_value != read_value) {
-    return kDifError;
-  }
+  // if (reg_value != read_value) {
```

Comment:
@a-will / @timothytrippel i think removing this check altogether is probably the right thing to do.
The pads each have "allowable" attributes, so if you try to enable an attribute that does not belong to a particular pad, it just won't show up.

In fpga, there are actually no pull-ups, so when I try to enable it the readback fails and that's what causes the `sram_ctrl_execution_test` to eventually die.

Let me know if you agree, if yes I can just remove this. If not, I can file an issue to track a better fix in the future and temporarily remove it.

Reply:
Hm... this "quick fix" ended up being more hacky than we thought, haha. From a certain perspective, we actually want to check that impossible requests are caught, as they may be critical to the test's function.

Originally, this was a little workaround for an assertion in DV, right? Shall we just check that we are on the DV "device" and only apply it there? Then I can follow up later with removing the pull-up here and disabling RX by default in the UART.

Let me know what you both think!

Reply:
hmmm... i see your point about knowing we tried to set something that wasn't possible.
okay let me make this part only for DV.  I guess longer term, when we have these things come up, we'll want to make them platform specific anyways.

Reply:
i actually had another PR pending that removed RX from test_rom and over to ottf, but that encountered some other kind of failure that i had not looked into, i'll table that for another PR.

Reply:
alright! updated and tests are (hopefully) passing.

Reply:
> Just to make sure, do you mean enable pull-up only on DV? But eventually we will probably have other portable code that enables pull-ups right? Ie, code that runs on both asic and fpga platforms. How would we want to handle those?

Yup, enable in DV only. Keep in mind that code that requires us to turn on internal pull-ups is _not_ portable across the ASIC and FPGA platforms. The FPGA does not support them.
 
> My original thinking was to move the attribute check out of the DIF and into the DIF caller, and have the DIF caller be responsible for the differences.

My thoughts there:
Because we must be enabling the pull-ups for a reason, it's best for us to ensure the lack of support is checked. It's easier to make that guarantee if the check is in the DIFs. If there is a reason the ASIC needs them enabled that is specific to the ASIC, then we can check against the device type and act accordingly.